packages/scene-composer/src/store/helpers/updateEntityHelper.ts

```diff
@@ -0,0 +1,50 @@
+import { UpdateEntityCommandInput } from '@aws-sdk/client-iottwinmaker';
+
+import { getGlobalSettings } from '../../common/GlobalSettings';
+import { IDataOverlayComponent, KnownComponentType } from '../../interfaces';
+import { updateNodeEntityComponent } from '../../utils/entityModelUtils/nodeComponent';
+import { updateOverlayEntityComponent } from '../../utils/entityModelUtils/overlayComponent';
+import { updateTagEntityComponent } from '../../utils/entityModelUtils/tagComponent';
+import { ISceneComponentInternal, ISceneNodeInternal } from '../internalInterfaces';
+
+export const updateEntity = async (
+  compToBeUpdated: ISceneComponentInternal | IDataOverlayComponent,
+  node: ISceneNodeInternal,
+  layerId: string,
+  type: string,
+): Promise<void> => {
+  const sceneMetadataModule = getGlobalSettings().twinMakerSceneMetadataModule;
+
+  if (!node.properties.matterportId) {
+    return;
+  }
+
+  if (!layerId) {
+    return;
+  }
+  const tagComp = type === 'Tag' ? updateTagEntityComponent(compToBeUpdated) : null;
+  const overlay =
+    type === 'DataOverlay' ? updateOverlayEntityComponent(compToBeUpdated as IDataOverlayComponent) : null;
```

Comment:
use switch statement
```
let comp: ISceneComponentInternal | undefined = undefined;
switch (compToBeUpdated.type) {
  case KnownComponentType.Tag:
    comp = updateTagEntityComponent(compToBeUpdated)
    break;
  .....
}
```
then below you just need to do 
```
comp ?? updateEntity.componentUpdates![comp.type] = comp;
```
it will be easier to support more types later

Reply:
We can't just assign updateTagEntityComponent as type of comp is ISceneComponentInternal. Which has other parameters too.